rules/Strict/Rector/If_/BooleanInIfConditionRuleFixerRector.php

```diff
@@ -83,45 +75,8 @@ public function getNodeTypes(): array
      */
     public function refactor(Node $node): ?If_
     {
-        $hasChanged = false;
-
-        $scope = ScopeFetcher::fetch($node);
-
-        // 1. if
-        $ifCondExprType = $scope->getNativeType($node->cond);
-        $notIdentical = $this->exactCompareFactory->createNotIdenticalFalsyCompare(
-            $ifCondExprType,
-            $node->cond,
-            $this->treatAsNonEmpty
+        throw new ShouldNotHappenException(
+            'This rule is deprecated and will be removed as risky and requires manual checking'
```

Comment:
we don't know what "this rule" is, should mention `self::class` instead.

Reply:
What's the output error? I think deprecated rules are reported on Rector run.

Reply:
I see, that seems fine then 👍